v0/destinations/hs/util.js

```diff
@@ -300,6 +302,142 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
   return payload;
 };
 
+const splitEventsForCreateUpdate = async (inputs, destination) => {
+  const { Config } = destination;
+  const values = [];
+  const updateHubspotIds = [];
+  const firstMessage = inputs[0].message;
+  let objectType = null;
+  let identifierType = null;
+  if (firstMessage) {
+    objectType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .objectType;
+    identifierType = getDestinationExternalIDInfoForRetl(firstMessage, "HS")
+      .identifierType;
+  }
+  inputs.map(async input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+    values.push(destinationExternalId);
+  });
+  let requestData = {
+    filterGroups: [
+      {
+        filters: [
+          {
+            propertyName: identifierType,
+            values,
+            operator: "IN"
+          }
+        ]
+      }
+    ],
+    properties: [identifierType],
+    limit: 1,
```

Comment:
why limit?

Reply:
actually we can set this too 1. Was testing say gave the value of 1. Will change.